components/files/configurationmanager.hpp

```diff
@@ -63,6 +65,53 @@ struct ConfigurationManager
         bool mSilent;
 };
 
+
+/**
+ * \struct escape_hash_filter
+ */
+struct escape_hash_filter : public boost::iostreams::input_filter
+{
+    static const int sEscape;
+    static const int sHashIdentifier;
+    static const int sEscapeIdentifier;
+
+    escape_hash_filter();
+    virtual ~escape_hash_filter();
+
+    template <typename Source> int get(Source & src);
+
+    private:
+        std::queue<int> mNext;
+        int mPrevious;
+
+        bool mSeenNonWhitespace;
+        bool mFinishLine;
+};
+
+/**
+ * \class EscapeHashString
+ */
+class EscapeHashString : public std::string
```

Comment:
Apparently I was still relying on a lexical cast, but that should be resolved pretty quickly.